apps/jobs/views.py

```diff
@@ -2775,3 +2775,45 @@ def update_submission_started_at(request, submission_pk):
         return Response(response_data, status=status.HTTP_200_OK)
     else:
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(["PATCH"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
+def update_submission_meta(request, challenge_pk, submission_pk):
+    """
+    Common API Endpoint for updating the submission meta data for hosts and participants.
+    """
+
+    if is_user_a_host_of_challenge(request.user, challenge_pk):
+        try:
+            submission = Submission.objects.get(
+                id=submission_pk,
+            )
+        except Submission.DoesNotExist:
+            response_data = {"error": "Submission does not exist"}
+            return Response(response_data, status=status.HTTP_403_FORBIDDEN)
+
+        serializer = SubmissionSerializer(
+            submission,
+            data=request.data,
+            context={
+                "request": request,
+            },
+            partial=True,
+        )
+
+        if serializer.is_valid():
+            serializer.save()
+            response_data = serializer.data
+            return Response(response_data, status=status.HTTP_200_OK)
+        else:
+            return Response(
+                serializer.errors, status=status.HTTP_400_BAD_REQUEST
+            )
+    else:
+        response_data = {
```

Comment:
I think we also discussed adding the `submission_cancel` feature in this API only, no?

Reply:
Let's make that change in a separate PR. I didn't want to merge them in a single one